installer/lib/daisy/single.ex

```diff
@@ -0,0 +1,349 @@
+
+  template(:text_input, [
+    {:eex, :web,
+     "components/text_input.ex": "lib/:lib_web_name/daisy_ui_components/text_input.ex"}
+  ])
+
+  template(:textarea, [
+    {:eex, :web, "components/textarea.ex": "lib/:lib_web_name/daisy_ui_components/textarea.ex"}
+  ])
+
+  template(:toggle, [
+    {:eex, :web, "components/toggle.ex": "lib/:lib_web_name/daisy_ui_components/toggle.ex"}
+  ])
+
+  template(:tooltip, [
+    {:eex, :web, "components/tooltip.ex": "lib/:lib_web_name/daisy_ui_components/tooltip.ex"}
+  ])
+
+  template(:dev, [
+    {:eex, :web, @all_components}
+  ])
+
+  def prepare_project(%Project{app: app, base_path: base_path} = project) when not is_nil(app) do
+    if in_umbrella?(base_path) do
+      %{project | in_umbrella?: true, project_path: Path.dirname(Path.dirname(base_path))}
+    else
+      %{project | in_umbrella?: false, project_path: base_path}
+    end
+    |> put_app()
+    |> put_root_app()
+    |> put_web_app()
+  end
+
+  defp put_app(%Project{base_path: base_path} = project) do
+    %{project | app_path: base_path}
+  end
+
+  defp put_root_app(%Project{app: app, opts: opts} = project) do
+    %{
+      project
+      | root_app: app,
+        root_mod: Module.concat([opts[:module] || Macro.camelize(app)])
+    }
+  end
+
+  defp put_web_app(%Project{app: app} = project) do
+    %{
+      project
+      | web_app: app,
+        lib_web_name: "#{app}_web",
+        web_namespace: Module.concat(["#{project.root_mod}Web"]),
+        web_path: project.base_path
+    }
+  end
+
+  def generate(%Project{} = project) do
+    all_components = all_components()
+
+    if Project.dev?(project) do
+      component_bindings =
+        all_components
+        |> then(fn list -> ["jS_helpers" | list] end)
+        |> Enum.sort()
+
+      project = %{project | binding: [{:components, component_bindings} | project.binding]}
+      copy_from(project, __MODULE__, :dev)
+    else
+      components = Keyword.get(project.opts, :components)
+
+      components =
+        if components == [] do
+          all_components
+        else
+          check_components(components, all_components)
+        end
+
+      requested_components_with_deps =
+        Enum.reduce(components, MapSet.new(["core"]), fn comp, acc ->
+          MapSet.put(acc, comp)
+          |> add_deps(comp)
+        end)
+
+      component_bindings =
+        Enum.reduce(requested_components_with_deps, [], fn x, acc ->
+          substitute_core(x, acc)
+        end)
+        |> Enum.sort()
+
+      project = %{project | binding: [{:components, component_bindings} | project.binding]}
+
+      requested_components_with_deps =
+        Enum.map(requested_components_with_deps, fn comp -> String.to_atom(comp) end)
+
+      Enum.each(requested_components_with_deps, fn component ->
+        copy_from(project, __MODULE__, component)
+      end)
+    end
+
+    project
+  end
+
+  defp substitute_core(x, acc) do
+    if x == "core", do: ["jS_helpers" | acc], else: [x | acc]
```

Comment:
is it "jS_helpers" or "js_helpers"

Reply:
yes, it is correct, the reason is because when you do `Macro.camelize` it transforms to `JSHelpers` instead of `JsHelpers`

Reply:
take a look to `templates/daisy_ui_components.ex` , it will make sense